clients/mobile/src/authentication/socialAuth/hooks/useSocialAuth.ts

```diff
@@ -0,0 +1,62 @@
+import { useState } from 'react'
+import { useNavigation, CommonActions, NavigationProp } from '@react-navigation/native'
+import { SCREENS } from '../../../setup/constant/screens'
+import { googleLogin, facebookLogin } from '../../services/authService'
+import { useAuth } from '../../context/useAuth'
+import { useFetchClient } from '../../../setup/clients/useFetchClient'
+import { useUserProfile } from '../../../userWorkflow/context/UserProfileContext'
+import registerUserDeviceForNotification from '../../../utility/deviceRegistration'
+import { SocialLoading } from '../types/loadingType'
+import { RootStackParamList } from '../../../setup/navigation/navigationTypes'
+import { SOCIAL_TYPES } from '../constants/socialTypes'
+
+interface UseSocialAuthOutput {
+  socialLoading: SocialLoading;
+  socialLoginError: string;
+  handleGoogleSignIn: () => Promise<void>;
+  handleFacebookSignIn: () => Promise<void>;
+}
+
+export const useSocialAuth = (): UseSocialAuthOutput => {
+  const fetchClient = useFetchClient()
+  const { userProfile } = useUserProfile()
+  const { setIsAuthenticated } = useAuth()
+  const [socialLoading, setSocialLoading] = useState<SocialLoading>('idle')
+  const [socialLoginError, setSocialLoginError] = useState<string>('')
+  const navigation = useNavigation<NavigationProp<RootStackParamList>>()
```

Comment:
We do have `LoginScreenNavigationProp` please use that one instead of `NavigationProp<RootStackParamList>`

Reply:
updated at 21f27f46ea6ee2109d40ca7599ba9feaabe7e59b